src/compositions/useLink.ts

```diff
@@ -149,4 +181,29 @@ function prefetchComponentsForRoute(route: ResolvedRoute, { routerPrefetch, link
     }
   })
 
+}
+
+type PropsPrefetchContext = {
+  routerPrefetch: PrefetchConfig | undefined,
+  linkPrefetch: PrefetchConfig | undefined,
+  getRouteProps: PropStore['getRouteProps'],
+}
```

Comment:
I'm hoping we can take `getRouteProps` off this type, and then this type is almost identical to `PrefetchConfigs` in types/prefetch.ts. Do we need both types? Even if we need it, we should move this to prefetch.ts and have the types build off each other